src/rdata/naptr.rs

```diff
@@ -0,0 +1,545 @@
+        match self.flags.partial_cmp(&other.flags) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.services.partial_cmp(&other.services) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.regexp.partial_cmp(&other.regexp) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+
+        Some(self.replacement.name_cmp(&other.replacement))
+    }
+}
+
+impl<Octs, OtherOcts, Name, OtherName>
+    CanonicalOrd<Naptr<OtherOcts, OtherName>> for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    OtherOcts: AsRef<[u8]>,
+    Name: ToName,
+    OtherName: ToName,
+{
+    fn canonical_cmp(&self, other: &Naptr<OtherOcts, OtherName>) -> Ordering {
+        match self.order.cmp(&other.order) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.preference.cmp(&other.preference) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.flags.canonical_cmp(&other.flags) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.services.canonical_cmp(&other.services) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.regexp.canonical_cmp(&other.regexp) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+
+        self.replacement.name_cmp(&other.replacement)
+    }
+}
+
+impl<Octs, Name> Ord for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    Name: ToName,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.order.cmp(&other.order) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.preference.cmp(&other.preference) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.flags.cmp(&other.flags) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.services.cmp(&other.services) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.regexp.cmp(&other.regexp) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+
+        self.replacement.name_cmp(&other.replacement)
+    }
+}
+
+//--- Hash
+
+impl<Octs, Name> hash::Hash for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    Name: hash::Hash,
+{
+    fn hash<H: hash::Hasher>(&self, state: &mut H) {
+        self.order.hash(state);
+        self.preference.hash(state);
+        self.flags.hash(state);
+        self.services.hash(state);
+        self.regexp.hash(state);
+        self.replacement.hash(state);
+    }
+}
+
+//--- RecordData, ParseRecordData, ComposeRecordData
+
+impl<Octs, Name> RecordData for Naptr<Octs, Name> {
+    fn rtype(&self) -> Rtype {
+        Rtype::NAPTR
```

Comment:
In other rdata implementations the style used here is `Naptr::RTYPE` rather than `Rtype::NAPTR`.

```suggestion
        Naptr::RTYPE
```